desktop/ui/src/main/java/org/datacleaner/panels/DataCloudInformationPanel.java

```diff
@@ -0,0 +1,185 @@
+
+    private static final int WIDTH = 360;
+    private static final int POSITION_Y = 130;
+
+    private final DCGlassPane _glassPane;
+    private final Color _background = WidgetUtils.BG_COLOR_BRIGHTEST;
+    private final Color _foreground = WidgetUtils.BG_COLOR_DARKEST;
+    private final Color _borderColor = WidgetUtils.BG_COLOR_MEDIUM;
+
+    private DCLabel text;
+    final DCHtmlBox htmlBoxDataCloud =
+            new DCHtmlBox("More information on <a href=\"http://datacleaner.org\">datacleaner.org</a>");
+
+    public DataCloudInformationPanel(DCGlassPane glassPane) {
+        super();
+        _glassPane = glassPane;
+        setBorder(new CompoundBorder(new LineBorder(_borderColor, 1), new EmptyBorder(20, 20, 20, 30)));
+        setVisible(false);
+        setSize(WIDTH, 400);
+        setLocation(getXWhenOut(), POSITION_Y);
+
+        setLayout(new VerticalLayout(10));
+
+        DCLabel header = DCLabel.darkMultiLine("DataCloud status");
+        header.setFont(WidgetUtils.FONT_HEADER1);
+        header.setIcon(ImageManager.get().getImageIcon("images/menu/datacloud.png"));
+        add(header);
+
+        text = DCLabel.darkMultiLine("");
+        add(text);
+        add(Box.createVerticalBox());
+        add(htmlBoxDataCloud);
+    }
+
+    private int getXWhenOut() {
+        return _glassPane.getSize().width + WIDTH + 10;
+    }
+
+    private int getXWhenIn() {
+        return _glassPane.getSize().width - WIDTH + 10;
+    }
+
+    public void setInformationStatus(RemoteServerState remoteServerState) {
+        text.setText("");
+        String panelContent = "";
+        if (remoteServerState.getActualState() == RemoteServerState.State.OK ||
+                remoteServerState.getActualState() == RemoteServerState.State.NO_CREDIT) {
+            panelContent = addLine(panelContent, "Connected as " + remoteServerState.getRealName());
+            panelContent = addLine(panelContent, "(email: " + remoteServerState.getEmail() + ")");
+            if (remoteServerState.getCredit() > 0) {
+                panelContent =
+                        addLine(panelContent, "Your credit balance: " + creditFormatter(remoteServerState.getCredit()));
+            } else {
+                panelContent = addLine(panelContent,
+                        "Your credit: <font color=\"red\">" + creditFormatter(remoteServerState.getCredit())
+                                + " </font>");
+            }
+
+            if (!remoteServerState.isEmailConfirmed()) {
+                panelContent = addLine(panelContent, "Email is not confirmed.");
+            }
+        }
+
+        if (remoteServerState.getActualState() == RemoteServerState.State.UNKNOWN) {
+            panelContent = addLine(panelContent, "Datacloud is not configured.");
+            panelContent = addLine(panelContent, "You can set your credentials in");
+            panelContent = addLine(panelContent, "<b>Options dialog - DataCloud</b>");
```

Comment:
All of this is very different from what we usually do. How come this isn't using standard Swing components instead of inline HTML?